gtsfm/averaging/translation/averaging_1dsfm.py

```diff
@@ -108,4 +123,97 @@ def run(
+    outlier_i1_i2_pairs: List[Tuple[int, int]],
+    i2Ui1_dict: Dict[Tuple[int, int], Optional[Unit3]],
+    wRi_list: List[Optional[Rot3]],
+    wti_list: List[Optional[Point3]],
+    gt_wTi_list: List[Optional[Pose3]],
+) -> GtsfmMetricsGroup:
+    """Computes the translation averaging metrics as a metrics group.
+
+    Args:
+        w_i2Ui1_inlier_measurements: Inlier 1DSfM translation direction measurements in world frame.
+        w_i2Ui1_outlier_measurements: Outlier 1DSfM translation direction measurements in world frame.
+        wRi_list: Estimated camera rotations from rotation averaging.
+        wti_list: Estimated camera translations from translation averaging.
+        gt_wTi_list: List of ground truth camera poses.
+
+    Returns:
+        Translation averaging metrics as a metrics group. Includes the following metrics:
+        - Number of inlier, outlier and total measurements.
+        - Distribution of translation direction angles for inlier measurements.
+        - Distribution of translation direction angle for outlier measurements.
+    """
+    # Get ground truth translation directions for the measurements.
+    gt_i2Ui1_dict = metrics_utils.get_twoview_translation_directions(gt_wTi_list)
+
+    # Angle between i2Ui1 measurement and GT i2Ui1 measurement for inliers and outliers.
+    inlier_measurement_angles = _get_measurement_angle_errors(inlier_i1_i2_pairs, i2Ui1_dict, gt_i2Ui1_dict)
+    outlier_measurement_angles = _get_measurement_angle_errors(outlier_i1_i2_pairs, i2Ui1_dict, gt_i2Ui1_dict)
+
+    measured_gt_i2Ui1_dict = {}
+    for (i1, i2) in inlier_i1_i2_pairs + outlier_i1_i2_pairs:
+        measured_gt_i2Ui1_dict[(i1, i2)] = gt_i2Ui1_dict[(i1, i2)]
+
+    # Compute estimated poses after the averaging step and align them to ground truth.
+    wTi_list = []
+    for (wRi, wti) in zip(wRi_list, wti_list):
+        if wRi is None or wti is None:
+            wTi_list.append(None)
+        else:
+            wTi_list.append(Pose3(wRi, wti))
+    wTi_aligned_list, _ = comp_utils.align_poses_sim3_ignore_missing(gt_wTi_list, wTi_list)
+
+    # Angle between relative estimated translations and ground truth measurements.
+    translation_angles = metrics_utils.compute_translation_angle_metric(measured_gt_i2Ui1_dict, wTi_aligned_list)
+
+    ta_metrics = []
+    num_total_measurements = len(inlier_i1_i2_pairs) + len(outlier_i1_i2_pairs)
+    ta_metrics.append(GtsfmMetric("num_total_1dsfm_measurements", num_total_measurements))
+    ta_metrics.append(GtsfmMetric("num_inlier_1dsfm_measurements", len(inlier_i1_i2_pairs)))
+    ta_metrics.append(GtsfmMetric("num_outlier_1dsfm_measurements", len(outlier_i1_i2_pairs)))
+    ta_metrics.append(GtsfmMetric("num_translations_estimated", len([wti for wti in wti_list if wti is not None])))
+    ta_metrics.append(GtsfmMetric("inlier_measurement_angles", inlier_measurement_angles))
+    ta_metrics.append(GtsfmMetric("outlier_measurement_angles", outlier_measurement_angles))
```

Comment:
i think i prefer a name like `xyz_angular_error_deg`, otherwise i cant tell if this is about the angles in the world frame or w.r.t. poses, or error